controllers/apps/operations/switchover.go

```diff
@@ -273,16 +249,46 @@ func handleSwitchoverProgress(reqCtx intctrlutil.RequestCtx, cli client.Client,
 		return expectCount, completedCount, failedCount, err
 	}
 
-	if completedCount == expectCount {
-		for _, jobName := range succeedJobs {
-			if err := job.CleanJobByName(reqCtx.Ctx, cli, opsRes.Cluster, jobName); err != nil {
-				reqCtx.Log.Error(err, "clean switchover job failed", "jobName", jobName)
-				return expectCount, completedCount, failedCount, err
-			}
-		}
+	return expectCount, completedCount, failedCount, nil
+}
+
+func doSwitchover(ctx context.Context, cli client.Reader, synthesizedComp *component.SynthesizedComponent,
+	switchover *appsv1alpha1.Switchover, switchoverCondition *metav1.Condition) error {
+	consistency, err := checkPodRoleLabelConsistency(ctx, cli, *synthesizedComp, switchover, switchoverCondition)
+	if err != nil {
+		return err
+	}
+	if consistency {
+		return nil
 	}
 
-	return expectCount, completedCount, failedCount, nil
+	pod, err := getServiceableNWritablePod(ctx, cli, *synthesizedComp)
+	if err != nil {
+		return err
+	}
+
+	pods, err := component.ListOwnedPods(ctx, cli, synthesizedComp.Namespace, synthesizedComp.ClusterName, synthesizedComp.Name)
+	if err != nil {
+		return err
+	}
+
+	lfa, err := lifecycle.New(synthesizedComp, pod, pods...)
```

Comment:
There is no need to specify a default pod, let the action define it yourself, or let the system select one randomly.

Reply:
> There is no need to specify a default pod, let the action define it yourself, or let the system select one randomly.

fixed